zerver/views/message_edit.py

```diff
@@ -125,6 +125,7 @@ def update_message_backend(
     send_notification_to_old_thread: bool = REQ(default=False, json_validator=check_bool),
     send_notification_to_new_thread: bool = REQ(default=True, json_validator=check_bool),
     content: Optional[str] = REQ(default=None),
+    check_time_limit_constraints: bool = REQ(default=False, json_validator=check_bool),
```

Comment:
You should start a thread in "api design" about how we want this API to work. I think the behavior we want long-term is that this endpoint, by default, will throw the new exception if you request a move that you don't have permission to fully complete -- a client is unlikely to actually want to do that. 

I think making partial moves that seem like they're not intended throw an error is a higher priority for me that avoiding backwards-compatibility, so I think we probably want something like `partial_move_if_limited` if we want a parameter at all. I think the other option would be for the exception to include the message ID of the first message that you do have permission to move, and a client that hits the "confirm" button should send a new request that's `change_later` with that message ID as the anchor, with no new API parameter.